JWBestPracticeApps/BasicVideoViewController/BasicVideoViewController.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  BasicVideoViewController.swift
+//  BasicVideoViewController
+//
+//  Created by David Almaguer on 9/2/19.
+//  Copyright © 2019 Karim Mourra. All rights reserved.
+//
+
+import UIKit
+
+let videoUrl = "http://playertest.longtailvideo.com/adaptive/oceans/oceans.m3u8"
+let thumbnailUrl = "http://d3el35u4qe4frz.cloudfront.net/bkaovAYt-480.jpg"
+
+class BasicVideoViewController: UIViewController {
+    
+    var player: JWPlayerController? = nil
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Instance JWPlayerController object
+        createPlayer()
+        if let containerView = view,
+            let playerView = player?.view {
+            containerView.addSubview(playerView)
+            // Turn off translatesAutoresizingMaskIntoConstraints property to use Auto Layout to dynamically calculate the size and position
+            playerView.translatesAutoresizingMaskIntoConstraints = false
+            // Add constraints to center the playerView
+            playerView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor).isActive = true
+            playerView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor).isActive = true
+            playerView.centerYAnchor.constraint(equalTo: containerView.centerYAnchor).isActive = true
+            playerView.heightAnchor.constraint(equalToConstant: 300).isActive = true
+        }
+    }
+    
+// MARK: Internal methods
+
+    func createPlayer() {
+        if let player = JWPlayerController(config: createConfig(), delegate: self as? JWPlayerDelegate) {
```

Comment:
instead of a soft casting, wouldn't it be more standard to have the view controller declare an extension to conform to the JWPlayerDelegate? In general we should avoid casting as much as possible.